core/txpool/blobpool/blobpool.go

```diff
@@ -0,0 +1,1461 @@
+			if filled && inclusions != nil {
+				p.offload(addr, txs[0].nonce, txs[0].id, inclusions)
+			}
+			txs = txs[1:]
+		}
+		log.Trace("Dropping overlapped blob transactions", "from", addr, "overlapped", nonces, "ids", ids, "left", len(txs))
+		for _, id := range ids {
+			if err := p.store.Delete(id); err != nil {
+				log.Error("Failed to delete blob transaction", "from", addr, "id", id, "err", err)
+			}
+		}
+		p.index[addr] = txs
+	}
+	// Iterate over the transactions to initialize their eviction thresholds
+	// and to detect any nonce gaps
+	txs[0].evictionExecTip = txs[0].execTipCap
+	txs[0].evictionExecFeeJumps = txs[0].basefeeJumps
+	txs[0].evictionBlobFeeJumps = txs[0].blobfeeJumps
+
+	for i := 1; i < len(txs); i++ {
+		// If there's no nonce gap, initialize the evicion thresholds as the
+		// minimum between the cumulative thresholds and the current tx fees
+		if txs[i].nonce == txs[i-1].nonce+1 {
+			txs[i].evictionExecTip = txs[i-1].evictionExecTip
+			if txs[i].evictionExecTip.Cmp(txs[i].execTipCap) > 0 {
+				txs[i].evictionExecTip = txs[i].execTipCap
+			}
+			txs[i].evictionExecFeeJumps = txs[i-1].evictionExecFeeJumps
+			if txs[i].evictionExecFeeJumps > txs[i].basefeeJumps {
+				txs[i].evictionExecFeeJumps = txs[i].basefeeJumps
+			}
+			txs[i].evictionBlobFeeJumps = txs[i-1].evictionBlobFeeJumps
+			if txs[i].evictionBlobFeeJumps > txs[i].blobfeeJumps {
+				txs[i].evictionBlobFeeJumps = txs[i].blobfeeJumps
+			}
+			continue
+		}
+		// Sanity check that there's no double nonce. This case would be a coding
+		// error, but better know about it
+		if txs[i].nonce == txs[i-1].nonce {
+			log.Error("Duplicate nonce blob transaction", "from", addr, "nonce", txs[i].nonce)
+		}
+		// Otherwise if there's a nonce gap evict all later transactions
+		var (
+			ids    []uint64
+			nonces []uint64
+		)
+		for j := i; j < len(txs); j++ {
+			ids = append(ids, txs[j].id)
+			nonces = append(nonces, txs[j].nonce)
+
+			p.spent[addr] = new(uint256.Int).Sub(p.spent[addr], txs[j].costCap)
+			p.stored -= uint64(txs[j].size)
+			delete(p.lookup, txs[j].hash)
+		}
+		p.index[addr] = txs[:i]
+
+		log.Warn("Dropping gapped blob transactions", "from", addr, "missing", txs[i-1].nonce+1, "drop", nonces, "ids", ids)
+		for _, id := range ids {
+			if err := p.store.Delete(id); err != nil {
+				log.Error("Failed to delete blob transaction", "from", addr, "id", id, "err", err)
+			}
+		}
+		break
+	}
+	// Ensure that there's no over-draft, this is expected to happen when some
+	// transactions get included without publishing on the network
+	var (
+		balance = uint256.MustFromBig(p.state.GetBalance(addr))
+		spent   = p.spent[addr]
+	)
+	if spent.Cmp(balance) > 0 {
+		// Evict the highest nonce transactions until the pending set falls under
+		// the account's available balance
+		var (
+			ids    []uint64
+			nonces []uint64
+		)
+		for p.spent[addr].Cmp(balance) > 0 {
+			last := txs[len(txs)-1]
```

Comment:
You are using the `txs` slice. But if there are some gapped txs, they will be removed here https://github.com/ethereum/go-ethereum/pull/26940/files#diff-a7bc0af0cfa11af9aa1a48112f0ebf710aefa78ad2bb1ba45a32aedc4cb8aaaeR595

However, you are still using the `txs` instead of `p.indexs[addr]`?